backend/script.sh

```
@@ -4,25 +4,182 @@ REPO_URL="https://github.com/AOSSIE-Org/EduAid.git"
+    
+    # Clone or update repository
+    if [ ! -d "$REPO_DIR" ]; then
+        echo -e "${GREEN}Cloning EduAid repository...${NC}"
+        git clone $REPO_URL
+        if [ $? -ne 0 ]; then
+            echo -e "${RED}Failed to clone repository${NC}"
+            exit 1
+        fi
+    else
+        echo -e "${YELLOW}EduAid repository already exists. Updating...${NC}"
+        cd $REPO_DIR
```

Shellcheck (0.11.0)

[warning] Use 'cd ... || exit' or 'cd ... || return' in case cd fails.

(SC2164)